browser/components/newtab/RemoteAboutNewTab.jsm

```diff
@@ -446,10 +456,10 @@ let RemoteAboutNewTab = {
           RemoteNewTabUtils.blockedLinks.resetCache();
           break;
       }
-    } else if (aTopic == "browser:purge-session-history") {
-        RemoteNewTabUtils.links.resetCache();
-        RemoteNewTabUtils.links.populateCache(() => {
-          this.pageListener.sendAsyncMessage("NewTab:UpdateLinks", {
+    } else if (aTopic === "browser:purge-session-history") {
```

Comment:
Nit: this `else if` is a bit ugly (particularly as it's not followed by a `} else {` ). Can it just be an normal if on its own on a new line?